auth/range_perm_cache.go

```diff
@@ -163,24 +163,35 @@ func checkKeyPerm(cachedPerms *unifiedRangePermissions, key, rangeEnd []byte, pe
 
 func (as *authStore) isRangeOpPermitted(tx backend.BatchTx, userName string, key, rangeEnd []byte, permtyp authpb.Permission_Type) bool {
 	// assumption: tx is Lock()ed
```

Comment:
Ah, yes, `be.BatchTx()` is working as a mutex for `rangePermCache`, too. But it isn't intended to protect `rangePermCache`, so adding the new mutex would make the intention clear and will make future change safer.

I think both of adding the new mutex or not is ok. Do you have an opinion about this?